src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/ECDsaCng.cs

```diff
@@ -29,6 +47,35 @@ public ECDsaCng(int keySize)
             KeySize = keySize;
         }
 
+        public override int KeySize
+        {
+            get
+            {
+                if (_curve != null)
+                {
+                    // Load the key so we don't return 0
+                    var key = Key;
+                    Debug.Assert(_curve == null);
+                }
+
+                return base.KeySize;
+            }
+            set
+            {
+                if (KeySize == value)
+                {
+                    return;
+                }
+
+                Debug.Assert(_curve == null);
+
+                // Set the KeySize first so that an invalid value doesn't throw away the key
```

Comment:
This comment I believe was taken from RSAOpenSsl. I changed three locations to say:
// Set the KeySize before FreeKey so that an invalid value doesn't throw away the key